src/gui/socketapi.cpp

```diff
@@ -621,6 +621,41 @@ void SocketApi::command_DOWNLOAD_VIRTUAL_FILE(const QString &filesArg, SocketLis
     }
 }
 
+/* Go over all the files ans replace them by a virtual file */
+void SocketApi::command_REPLACE_VIRTUAL_FILE(const QString &filesArg, SocketListener *)
+{
+    QStringList files = filesArg.split(QLatin1Char('\x1e')); // Record Separator
+    auto suffix = QStringLiteral(APPLICATION_DOTVIRTUALFILE_SUFFIX);
+
+    for (const auto &file : files) {
+        QString relativePath;
+        auto folder = FolderMan::instance()->folderForPath(file, &relativePath);
+        if (!folder)
+            continue;
+        if (file.endsWith(suffix))
+            continue;
+        QFileInfo fi(file);
+        if (fi.isDir()) {
+            folder->journalDb()->getFilesBelowPath(relativePath.toUtf8(), [&](const SyncJournalFileRecord &rec) {
+                if (rec._type != ItemTypeFile || rec._path.endsWith(APPLICATION_DOTVIRTUALFILE_SUFFIX))
+                    return;
+                QString file = folder->path() + '/' + QString::fromUtf8(rec._path);
+                if (!QFile::rename(file, file + suffix)) {
+                    qCWarning(lcSocketApi) << "Unable to rename " << file;
+                }
+            });
+            continue;
+        }
+        SyncJournalFileRecord record;
+        if (!folder->journalDb()->getFileRecord(relativePath, &record) || !record.isValid())
+            continue;
+        if (!QFile::rename(file, file + suffix)) {
+            qCWarning(lcSocketApi) << "Unable to rename " << file;
+        }
+        FolderMan::instance()->scheduleFolder(folder);
```

Comment:
Probably should not happen inside the loop

Reply:
yeah...  but this will be queued annyway.  And folder might be different (altough it shouldn't)